Proxy.h

```diff
@@ -54,14 +56,14 @@ class ProxyTracker
 	};
 
 	LinkedProxy* head;
-	pthread_mutex_t lock;
+	std::mutex lock;
 };
 
 template <class RefAllocator, class ...TX>
 class ProxyGenerator : public ProxyObject, public TX::__Proxy...
 {	
 protected:
-	ProxyGenerator() : m_ProxyObject(NewInstance(this, (jobject[]){TX::__CLASS...}, sizeof...(TX)))	
+	ProxyGenerator() : m_ProxyObject(NewInstance(this, TX::__CLASS...))
```

Comment:
This last change is not correct. By it you limit the number of interfaces implemented by proxy to 2.
You need to pass nullptr as additional argument after "TX::__CLASS..." and have:
static jobject NewInstance(void* nativePtr, ...);
where you would use va_list to go over extra args until you encounter nullptr.

Reply:
Alternatively it can be a recursive variadic function template that fills an array. But I personally would go with va_list instead.

Reply:
I think I have even better variant, no recursion and compiles on Windows and Mac

https://github.com/Unity-Technologies/android-jni-bridge/commit/5b1486174506d78c4605fd1c9bdd63f8e86d5907